src/logger.ts

```diff
@@ -415,9 +414,7 @@ export class Logger {
       await fs.promises.access(logFile, fs.constants.W_OK);
     } catch (err1) {
       try {
-        await mkdirp(path.dirname(logFile), {
-          mode: '700',
-        });
+        await fs.promises.mkdir(path.dirname(logFile), { recursive: true, mode: 0x700 });
```

Comment:
Make it platform safe here too

Reply:
When using recursive === true, the operation will not throw if the path already exists, so the catch should not be a `noop`;

See https://nodejs.org/api/fs.html#fspromisesmkdirpath-options